app/models/user/settings.rb

```diff
@@ -0,0 +1,25 @@
+User::SETTINGS = [
+  :new_submission_notifications
+]
+
+User::DEFAULT_SETTINGS = {
+  new_submission_notifications: true
+}
+
+module User::Settings
+  extend ActiveSupport::Concern
+
+  included do
+    scope :with_setting_not_disabled, ->(setting) { where("settings ->> ? IS NULL OR settings ->> ? != 'false'", setting.to_s, setting.to_s) }
```

Comment:
The current code assumes `NULL` to be a falsy value. In reality, we likely want `NULL` to be the default value.

---

I think we might want:

```suggestion
    scope :with_setting_enabled, ->(setting) { where("COALESCE(settings ->> ?, ?) == 'true'", setting.to_s, User::DEFAULT_SETTINGS[setting]) }
```

or

```suggestion
    scope :with_setting, ->(setting, value) { where("COALESCE(settings ->> ?, ?) == ?", setting.to_s, User::DEFAULT_SETTINGS[setting], value) }
```

I haven't tested this code, but `COALESCE` should allow us to apply the default settings even if the database value is `NULL`.

---

side note: I wonder if `User::DEFAULT_SETTINGS[setting]` needs to be serialized as JSONB before being passed into the query above.